test/helpers/http-server.ts

```diff
@@ -0,0 +1,157 @@
+import http = require('http');
+import https = require('https');
+import net = require('net');
+import express = require('express');
+import pify = require('pify');
+import pem = require('pem');
+import bodyParser = require('body-parser');
+import tempy = require('tempy');
+
+export type BaseServerOptions = {
+	protocol: 'HTTP' | 'HTTPS' | 'socket';
+	installBodyParser?: boolean;
+};
+export type HttpServerOptions = BaseServerOptions & {
+	protocol: 'HTTP';
+};
+export type HttpsServerOptions = BaseServerOptions & {
+	protocol: 'HTTPS';
+	commonName?: string;
+	days?: number;
+};
+export type HttpSocketServerOptions = BaseServerOptions & {
+	protocol: 'socket';
+};
+export type ServerOptions = HttpServerOptions | HttpsServerOptions | HttpSocketServerOptions;
+
+export interface BaseExtendedServer extends express.Express {
+	close: () => Promise<any>;
```

Comment:
This should be better typed than `any`.

Reply:
Actually, I put `any` because I don't care of the result, maybe `unknown` is better?

Reply:
`Promise<void>`